File: leetcode/0167-two-sum-ii---input-array-is-sorted/solution.java
class Solution {
    public int[] twoSum(int[] num, int t) {
        int l =0,r=num.length-1;

        while(l<r){
            int sum = num[l]+num[r];
            if(sum > t){
                r--;
            } else if(sum<t){
                l++;
            }else{
                return new int[] {l+1,r+1};
            }
        }

        return new int[] {-1,-1};
    }
}

File: leetcode/0167-two-sum-ii---input-array-is-sorted/solution.py
class Solution(object):
    def twoSum(self, numbers, target):
        ans = [0,0]

        l,r = 0 ,len(numbers)-1

        while l<r:
            added = numbers[l]+numbers[r]
            if added > target:
                r -= 1
            elif added < target:
                l += 1
            else:
                ans[0] = l+1
                ans[1] = r+1
                break
        return ans
        
